node/actors/network/src/lib.rs

```diff
@@ -131,12 +131,7 @@ impl Runner {
             });
 
             // Update QC batches in the background.
-            s.spawn(async {
-                match self.net.gossip.run_batch_qc_finder(ctx).await {
-                    Err(ctx::Error::Canceled(_)) => Ok(()),
-                    other => other,
-                }
-            });
+            s.spawn(self.net.gossip.run_batch_qc_finder(ctx));
```

Comment:
Okay, I understand now why it behaved differently: a `.context` wrapped a `Cancel` into an `anyhow::Error` which then became a `ctx::Error::Internal` and caused the scope to fail during shutdown. 

This is a bit obscure; it would be helpful if `Wrap` or at least the [implementation of it](https://github.com/matter-labs/era-consensus/blob/cf01b6add5cb8993411b29bdc1ba2627bb993e0f/node/libs/concurrency/src/ctx/mod.rs#L269) would mention the significance of consistently using `.wrap` with `ctx::Result`.

Reply:
Good point, I've added documentation to both Wrap and ctx::Error to cover that. If you think it is still too obscure, we can for example consider removing implicit conversion from anyhow::Error to ctx::Error.